staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1155,27 +1160,11 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 		listOptions = listOptions.DeepCopy()
 	}
 
-	listObj, err := e.List(ctx, listOptions)
-	if err != nil {
-		return nil, err
-	}
-	items, err := meta.ExtractList(listObj)
-	if err != nil {
-		return nil, err
-	}
-	if len(items) == 0 {
-		// Nothing to delete, return now
-		return listObj, nil
-	}
-	// Spawn a number of goroutines, so that we can issue requests to storage
-	// in parallel to speed up deletion.
-	// It is proportional to the number of items to delete, up to
-	// DeleteCollectionWorkers (it doesn't make much sense to spawn 16
-	// workers to delete 10 items).
+	itemsLock := sync.RWMutex{}
+	var items []runtime.Object
```

Comment:
items is growing indefinitely, correct? Technically this is unnecessary if we used it as a queue with some limited size or so. Worth it?

Reply:
Unfortunately it's needed because we return it at the end...

Given that (and given it's bounded by the size of all objects of a given type), I don't think we want to change it (at least for now).

Reply:
ic. Unfortunate.